design-docs/src/mdbook/src/dataplane/design-session.md

```diff
@@ -0,0 +1,393 @@
+
+@enduml
+```
+
+> Map of the relationships between planned dataplane components
+
+</figure>
+
+<section>
+
+### Configuration Store
+
+I could (and maybe should) write a book about the design considerations of [Configuration Store].
+For the moment I will limit myself to a list of hard and fast requirements:
+
+1. CP in the [CAP theorem](https://en.wikipedia.org/wiki/CAP_theorem) sense.
+  - immediate consistency in the sense that
+
+    > Every read receives either the most recent data or an error.
+
+  - partition tolerance
+
+    > The system continues to operate despite an arbitrary number of messages being dropped (or delayed) by the network between nodes.
+
+   The guiding theory is that
+
+  1. It is better to **not** function than to **mal**function.
+  2. _**It doesn't matter how quickly you can do the wrong thing**_.
+
+</section>
+<section>
+
+### Gateway Agent
+
+This is another subject deserving of a small book.
+
+For now, I will point out some notable design decisions we need to make:
+
+1. Do we expect a subscription model?
+2. If not, do we expect the [gateway agent] to explicitly push state to dependent components?
+
+Beyond that, we need to make some high-level design choices:
+
+1. programming language? Likely Go or Rust.
+2. REST? GraphQL? I tend to think REST is more appropriate at this time.
+
+</section>
+<section>
+
+### FRR agent
+
+Be afraid. Make Fredi fill in this section. But also be afraid.
+
+</section>
+<section>
+
+### Zebra Plugin
+
+This is a planned [zebra] plugin in the same spirit as [`fpm`](https://docs.frrouting.org/projects/dev-guide/en/latest/fpm.html#id1) or [`dataplane_fpm_nl`](https://docs.frrouting.org/projects/dev-guide/en/latest/fpm.html#dplane-fpm-nl).
+
+The core idea is to have a plugin that can be dynamically loaded into [zebra] and will listen to the [zebra event stream](https://github.com/FRRouting/frr/blob/ee5a3456d34a756c70ad8856ab7be7bed75ee31c/zebra/zebra_dplane.h#L114-L217) for updates.
+The plugin will then take those updates and push them into the dataplane agent, allowing the dataplane to react to route updates.
+
+</section>
+<section>
+
+### Routing daemons
+
+For the moment these are [`bgpd`](https://docs.frrouting.org/en/latest/bgp.html) and [`bfdd`](https://docs.frrouting.org/en/latest/bfd.html).
+
+</section>
+<section>
+
+### Interface Manager
+
+This is a component that exchanges [netlink] messages with the [kernel] in response to changes in the [dataplane model].
+Its responsibilities include
+
+1. construction of virtual network interfaces needed by [zebra]
+2. translation of ephemeral linux kernel parameters into ephemeral [dpdk] parameters (e.g. netlink interface index to dpdk interface id).
+3. retrieval of information not available to [zebra]/[frr] such as neighbor tables / [ARP] / [IPv6 ND] resolution or [bridge] fdb.
+
+### Control Plane Interface
+
+This component is responsible for adjudicating communication between the [control plane] and the [dataplane].
```

Comment:
The `[dataplane]` is referenced multiple times, but it doesn't seem to have a corresponding entry; the links don't take me anywhere

Reply:
yup, will fix